src/Core/Domain/Product/Pack/Command/SetPackProductsCommand.php

```diff
@@ -90,11 +90,10 @@ private function setProducts(array $products): void
 
         foreach ($products as $product) {
             $this->assertQuantity($product['quantity']);
-
             $this->products[] = new QuantifiedProduct(
-                $product['product_id'],
-                $product['quantity'],
-                isset($product['combination_id']) ? $product['combination_id'] : null
+                (int) $product['product_id'],
+                (int) $product['quantity'],
+                (int) $product['combination_id'] ?? null
```

Comment:
This expression might cause errors, or at least unexpected results. Since priority of casting operator is higher than coalescence, integer casting is executed first. This will crash execution if the array value is undefined.
```suggestion
                (int) ($product['combination_id'] ?? null)
```

Reply:
I apply it.